etcdutl/snapshot/v3_snapshot.go

```diff
@@ -265,6 +276,19 @@ func (s *v3Manager) Restore(cfg RestoreConfig) error {
 	if err = s.saveDB(); err != nil {
 		return err
 	}
+
+	if cfg.RevisionBump > 0 {
```

Comment:
Both bumpRevision and markLatestRevCompacted should be done in one transaction so it can be reverted on error.